### ISSUE TITLE
Update CNV WDLs to WDL 1.0.

### ISSUE THREAD

**Issue by samuelklee** [broadinstitute/gatk#6502]

**Comment by ldgauthier** [comment]
Are these `NUM_INTERVALS=${dollar}(grep -v '@' ${interval_list} | wc -l)` shenanigans in ScatterIntervals solved in WDL 1.0?

**Comment by samuelklee** [comment]
If you're talking about the need for `dollar`, I'm not sure---looks like there are other instances of this hack in other WDLs, some of which have purportedly been updated to 1.0.

I don't think we would need this hack (in the gCNV WDLs, at least) if it weren't for the bash-related shenanigans in that task, which are required due to the wonky output style of IntervalListTools, though...

**Comment by ldgauthier** [comment]
We can certainly modify IntervalListTools to make the behavior more intuitive (e.g. add a new mode for `INTERVAL_COUNT_WITH_OVREFLOW`), but I'm not sure I understand the issue.  We're just trying to avoid calling the GATK command if the scatter is going to be a noop?

The GATK SplitIntervals has slightly different behavior that's helpful in some cases.

**Comment by ldgauthier** [comment]
From https://github.com/gatk-workflows/gatk4-germline-cnvs/pull/2
`PreprocessIntervals` doesn't need a sequence dictionary arg if it has a reference fasta

**Comment by samuelklee** [comment]
We already added the functionality needed for the gCNV workflow to IntervalListTools in https://github.com/broadinstitute/picard/pull/1208.  The issue is that it outputs each scattered interval list to a separate directory if the number of scatters is greater than 1, but it just outputs to a file if we don't need to scatter.  Not sure the reason for this design, but it makes things difficult from the perspective of WDL.  Would be easier if the expected output was always `Array[File]+`.

I don't really see why IntervalListTools needs to create those intermediate directories (nor why it needs to be limited to number of scatters <= 10000 by the naming scheme, which limits the scatter index to 4 digits---don't think this is documented, either), but I am not sure if that behavior is expected by now or if it is safe to modify it.

Pretty sure SplitIntervals is just calling the same backend class used by IntervalListTools.  Perhaps that tool might've been spun off before we exposed the Picard tools?  See e.g. https://github.com/broadinstitute/gatk/pull/5392#issuecomment-435588845.  I think we should try to avoid writing custom GATK tools wh
